src/compute/src/sink/persist_sink.rs

```diff
@@ -136,14 +140,18 @@ where
             let mut stash: HashMap<_, Vec<_>> = HashMap::new();
 
             // TODO(aljoscha): We need to figure out what to do with error results from these calls.
-            let client = persist_clients
+            let persist_client = persist_clients
                 .lock()
                 .await
                 .open(persist_location)
                 .await
                 .expect("could not open persist client");
 
-            let mut write = client
+            if truncate {
```

Comment:
sorry if I'm just being blind, but do we also need to limit this to one worker?

Reply:
Ah, damn. You are right!